### PR TITLE
Fix event type,topic,subtopic dropdowns in event wizard

### LINKED ISSUE #604: Event topic, subTopic and type don't get displayed in event edit wizard dropdowns
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

**Current behavior:**

Event topic, subTopic and type don't get displayed in event edit wizard dropdowns

**Expected behavior:**

They should load data from event model
image

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Fixes event type,topic,subtopic dropdowns in event wizard

Fixes #604

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-605-open-event-frontend-fossasia.surge.sh

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 1]
```diff
@@ -446,7 +446,7 @@
 
   <div class="field">
     <label>{{t 'Event Type'}}</label>
-    {{#ui-dropdown class='search selection' selected=data.event.type onChange=(action (mut data.event.type)) as |execute mapper|}}
+    {{#ui-dropdown class='search selection' allowAdditions=true selected=data.event.type.name onChange=(action (mut data.event.type)) as |execute mapper|}}
```
Comment (niranjan94) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: We do not allow additionsFor this. The user must not be able to add their own type.

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 2]
```diff
@@ -458,7 +458,7 @@
   </div>
   <div class="field">
     <label>{{t 'Event Topic'}}</label>
-    {{#ui-dropdown class='search selection' selected=data.event.topic onChange=(action (mut data.event.topic)) as |execute mapper|}}
+    {{#ui-dropdown class='search selection' allowAdditions=true selected=data.event.topic.name onChange=(action (mut data.event.topic)) as |execute mapper|}}
```
Comment (niranjan94) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: We do not allow additionsFor this. The user must not be able to add their own topic.

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 3]
```diff
@@ -470,7 +470,7 @@
   </div>
   <div class="field">
     <label>{{t 'Event Sub-topic'}}</label>
-    {{#ui-dropdown class='search selection' selected=data.event.subTopic onChange=(action (mut data.event.subTopic)) as |execute mapper|}}
+    {{#ui-dropdown class='search selection' allowAdditions=true selected=data.event.subTopic.name onChange=(action (mut data.event.subTopic)) as |execute mapper|}}
```
Comment (niranjan94) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: We do not allow additionsFor this. The user must not be able to add their own subtopic.

**Comment by geekyd** [comment]
@niranjan94 Without `allowAdditions=true` the names of topic, type & subtopic is not getting rendered.

**Comment by CosmicCoder96** [comment]
@niranjan94 @geekyd 
> Without allowAdditions=true the names of topic, type & subtopic is not getting rendered.

Yes, I cannot get them to render without `allowAdditions` could this be a bug with semantic ui - ember ?

**Comment by niranjan94** [comment]
@CosmicCoder96 @geekyd could be a bug. Need to check. But lets try a workaround. 

- Don't pass anything to `selected`
- Instead, show the current selected one in the `default text` div. When nothing is selected, show, `Select <whatever>`.

Because, we should not allow additions for this.

**Comment by CosmicCoder96** [comment]
@niranjan94 Sounds like a good idea, except that the default text will have a separate styling. 
image

**Comment by niranjan94** [comment]
@CosmicCoder96 that's okay ... Thats something we can live with for now :)

**Comment by niranjan94** [comment]
Deployment link: https://pr-605-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-605-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@niranjan94 Done !!

**Pull request merged**